packages/shared/src/utils/generate-joi.ts

```diff
@@ -113,23 +115,31 @@ export function generateJoi(filter: FieldFilter, options?: JoiOptions): AnySchem
 		}
 
 		if (operator === '_contains') {
-			if (compareValue === null || compareValue === undefined) {
+			if (compareValue === null || compareValue === undefined || typeof compareValue !== 'string') {
+				schema[key] = Joi.any().equal(true);
+			} else {
+				schema[key] = getStringSchema().contains(compareValue);
+			}
+		}
+
+		if (operator === '_icontains') {
+			if (compareValue === null || compareValue === undefined || typeof compareValue !== 'string') {
 				schema[key] = Joi.any().equal(true);
 			} else {
 				schema[key] = getStringSchema().contains(compareValue);
 			}
 		}
 
 		if (operator === '_ncontains') {
-			if (compareValue === null || compareValue === undefined) {
+			if (compareValue === null || compareValue === undefined || typeof compareValue !== 'string') {
```

Comment:
I don't think this compareValue check should happen here 🤔 feels weird to silently fail the validation if the wrong type was presented. If anything, it should probably crash with an InvalidPayloadException when an illegal value is presented